appservice/src/SiteWrapper.ts

```diff
@@ -537,7 +547,9 @@ export class SiteWrapper {
             await new Promise((resolve: () => void): void => { setTimeout(resolve, pollingInterval); });
             deployment = await kuduClient.deployment.getResult(deploymentId);
         }
-
+        if (deployment.id === previousDeploymentId) {
```

Comment:
```
if (previousDeploymentId !== undefined && deployment.id === previousDeploymentId)
```

Reply:
Is this necessary?  If the previousDeploymentId is undefined, this will not be true unless deployment.id is also undefined which would probably be a failure anyway.

Reply:
True - I didn't think about it that way. I might still prefer it for the sake of readability, but no strong preference

Reply:
👑